File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/dataset/search/PhysicalDataSearchCriteria.java
/*
 * Copyright 2016 ETH Zuerich, CISD
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ch.ethz.sis.openbis.generic.asapi.v3.dto.dataset.search;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search.AbstractCompositeSearchCriteria;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search.SearchCriteriaToStringBuilder;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author pkupczyk
 */
@JsonObject("as.dto.dataset.search.PhysicalDataSearchCriteria")
public class PhysicalDataSearchCriteria extends AbstractCompositeSearchCriteria
{

    private static final long serialVersionUID = 1L;

    public ShareIdSearchCriteria withShareId()
    {
        return with(new ShareIdSearchCriteria());
    }

    public LocationSearchCriteria withLocation()
    {
        return with(new LocationSearchCriteria());
    }

    public SizeSearchCriteria withSize()
    {
        return with(new SizeSearchCriteria());
    }

    public StorageFormatSearchCriteria withStorageFormat()
    {
        return with(new StorageFormatSearchCriteria());
    }

    public FileFormatTypeSearchCriteria withFileFormatType()
    {
        return with(new FileFormatTypeSearchCriteria());
    }

    public LocatorTypeSearchCriteria withLocatorType()
    {
        return with(new LocatorTypeSearchCriteria());
    }

    public CompleteSearchCriteria withComplete()
    {
        return with(new CompleteSearchCriteria());
    }

    public StatusSearchCriteria withStatus()
    {
        return with(new StatusSearchCriteria());
    }

    public PresentInArchiveSearchCriteria withPresentInArchive()
    {
        return with(new PresentInArchiveSearchCriteria());
    }

    public StorageConfirmationSearchCriteria withStorageConfirmation()
    {
        return with(new StorageConfirmationSearchCriteria());
    }

    public SpeedHintSearchCriteria withSpeedHint()
    {
        return with(new SpeedHintSearchCriteria());
    }

    @Override
    protected SearchCriteriaToStringBuilder createBuilder()
    {
        SearchCriteriaToStringBuilder builder = super.createBuilder();
        builder.setName("PHYSICAL_DATA");
        return builder;
    }

}

File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/dataset/search/PhysicalDataSearchCriteria.js
/**
 * @author pkupczyk
 */
define([ "require", "stjs", "as/dto/common/search/AbstractCompositeSearchCriteria", "as/dto/dataset/search/ShareIdSearchCriteria", "as/dto/dataset/search/LocationSearchCriteria",
		"as/dto/dataset/search/SizeSearchCriteria", "as/dto/dataset/search/StorageFormatSearchCriteria", "as/dto/dataset/search/FileFormatTypeSearchCriteria",
		"as/dto/dataset/search/LocatorTypeSearchCriteria", "as/dto/dataset/search/CompleteSearchCriteria", "as/dto/dataset/search/StatusSearchCriteria",
		"as/dto/dataset/search/PresentInArchiveSearchCriteria", "as/dto/dataset/search/StorageConfirmationSearchCriteria", "as/dto/dataset/search/SpeedHintSearchCriteria" ], function(require, stjs,
		AbstractCompositeSearchCriteria) {
	var PhysicalDataSearchCriteria = function() {
		AbstractCompositeSearchCriteria.call(this);
	};
	stjs.extend(PhysicalDataSearchCriteria, AbstractCompositeSearchCriteria, [ AbstractCompositeSearchCriteria ], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.dataset.search.PhysicalDataSearchCriteria';
		constructor.serialVersionUID = 1;
		prototype.withShareId = function() {
			var ShareIdSearchCriteria = require("as/dto/dataset/search/ShareIdSearchCriteria");
			return this.addCriteria(new ShareIdSearchCriteria());
		};
		prototype.withLocation = function() {
			var LocationSearchCriteria = require("as/dto/dataset/search/LocationSearchCriteria");
			return this.addCriteria(new LocationSearchCriteria());
		};
		prototype.withSize = function() {
			var SizeSearchCriteria = require("as/dto/dataset/search/SizeSearchCriteria");
			return this.addCriteria(new SizeSearchCriteria());
		};
		prototype.withStorageFormat = function() {
			var StorageFormatSearchCriteria = require("as/dto/dataset/search/StorageFormatSearchCriteria");
			return this.addCriteria(new StorageFormatSearchCriteria());
		};
		prototype.withFileFormatType = function() {
			var FileFormatTypeSearchCriteria = require("as/dto/dataset/search/FileFormatTypeSearchCriteria");
			return this.addCriteria(new FileFormatTypeSearchCriteria());
		};
		prototype.withLocatorType = function() {
			var LocatorTypeSearchCriteria = require("as/dto/dataset/search/LocatorTypeSearchCriteria");
			return this.addCriteria(new LocatorTypeSearchCriteria());
		};
		prototype.withComplete = function() {
			var CompleteSearchCriteria = require("as/dto/dataset/search/CompleteSearchCriteria");
			return this.addCriteria(new CompleteSearchCriteria());
		};
		prototype.withStatus = function() {
			var StatusSearchCriteria = require("as/dto/dataset/search/StatusSearchCriteria");
			return this.addCriteria(new StatusSearchCriteria());
		};
		prototype.withPresentInArchive = function() {
			var PresentInArchiveSearchCriteria = require("as/dto/dataset/search/PresentInArchiveSearchCriteria");
			return this.addCriteria(new PresentInArchiveSearchCriteria());
		};
		prototype.withStorageConfirmation = function() {
			var StorageConfirmationSearchCriteria = require("as/dto/dataset/search/StorageConfirmationSearchCriteria");
			return this.addCriteria(new StorageConfirmationSearchCriteria());
		};
		prototype.withSpeedHint = function() {
			var SpeedHintSearchCriteria = require("as/dto/dataset/search/SpeedHintSearchCriteria");
			return this.addCriteria(new SpeedHintSearchCriteria());
		};
	}, {
		criteria : {
			name : "Collection",
			arguments : [ "ISearchCriteria" ]
		}
	});
	return PhysicalDataSearchCriteria;
})